src/pykeen/models/predict.py

```diff
@@ -51,6 +52,106 @@
 logger = logging.getLogger(__name__)
 
 
+@dataclasses.dataclass
+class Predictions:
+    """Base class for predictions"""
+
+    #: the dataframe; has to have a column named "score"
```

Comment:
you can put a `__post_init__` that raises an error if this isn't true

Reply:
https://github.com/pykeen/pykeen/pull/1090/commits/6e4a273319f5f6c1aec3f424cff0edaede18567b